### ISSUE TITLE
Reversed `assert_with_ulp` arguments in TTNN tests

### ISSUE THREAD

**Issue by jasondavies** [tenstorrent/tt-metal#55657]
# Summary

Several tests pass device output first and the reference second to
`assert_with_ulp(expected_result, actual_result, ...)`.
This comparison is asymmetric: it measures `abs(actual - expected) / ULP(expected)`
using the actual tensor's dtype for the ULP spacing. Reversing the arguments can
therefore hide accuracy regressions or cause false failures, particularly across
power-of-two boundaries. This is a call-site bug, not a request to change the metric.

## Minimal reproducer

Run in the repository's Python environment; no accelerator is needed:

```python
import torch
from tests.ttnn.utils_for_testing import assert_with_ulp

expected = torch.tensor([0.9999998807907104], dtype=torch.float32)
actual = torch.tensor([1.0], dtype=torch.float32)

assert_with_ulp(actual, expected, ulp_threshold=1)  # Incorrect order: passes (1 ULP).
assert_with_ulp(expected, actual, ulp_threshold=1)  # Correct order: fails (2 ULP).
```

This behavior was reproduced on the host; the analogous boundary case also
reproduces with BF16, FP16, and FP64.

## Affected tests

A repository-wide audit of 257 calls in 64 files found 51 reversed call sites.
All 51 also exist on `origin/main` at `e90ebd676b405a026ea30c5153b8bf79a99d9001`.

Paths below are relative to `tests/ttnn/`:

| File | Reversed calls |
| --- | ---: |
| `unit_tests/operations/eltwise/test_unary.py` | 16 |
| `unit_tests/operations/eltwise/test_unary_ops_ttnn.py` | 12 |
| `unit_tests/operations/eltwise/test_binary_int32.py` | 10 |
| `unit_tests/operations/eltwise/test_binary_composite.py` | 3 |
| `unit_tests/operations/eltwise/test_binaryng_ND.py` | 2 |
| `unit_tests/operations/eltwise/test_div_ops.py` | 2 |
| `unit_tests/operations/eltwise/test_unary_sharding.py` | 2 |
| `unit_tests/operations/eltwise/test_ternary.py` | 2 |
| `nightly/unit_tests/operations/eltwise/test_unary_shard.py` | 1 |
| `nightly/unit_tests/operations/eltwise/test_ternary_sharded.py` | 1 |

Of these, 41 are reachable, positive-tolerance comparisons. The other ten are
five dormant dtype branches, two zero-tolerance checks, and three checks followed
by exact equality assertions. Not every reversed call currently changes pass/fail.

## Proposed fix

- Put the reference first at the confirmed sites, preferably using explicit
  `expected_result=` and `actual_result=` keywords.
- Add a host-only regression covering the asymmetric power-of-two boundary.
- Rerun the affected device tests with their existing thresholds. Investigate
  any newly exposed failures rather than automatically relaxing tolerances.

Device-test impact has not yet been validated for the proposed swaps.
